### DISCUSSION TITLE
x-data Access from external scope

### DISCUSSION THREAD

**Discussion by amira-98** [alpinejs/alpine#1772] [1. Help]
Hello , 
I am trying to access x-data from an external Scope but document.querySelector('[x-data]').__x.getUnobservedData() return undefined if I use a simple div . Actually, it only works if I use a <template></template>

**Comment by Pronian** [comment]
```JS
document.querySelector('[x-data]')._x_dataStack[0]
```
This is what you are thinking of, but keep in mind that this is not the recommended way of getting data from a component. The recommended way is either using [$store](https://alpinejs.dev/magics/store) or via custom events with [$dispatch](https://alpinejs.dev/magics/dispatch)
